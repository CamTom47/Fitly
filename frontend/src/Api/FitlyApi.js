import axios from 'axios';

const BASE_URL = process.env.REACT_APP_BASE_URL || "http://localhost:3001";

/** Fitly API Class
 * 
 * Static class typing together methods used to get/send to the Fitly API.
 * There shouldn't be any frontend-specific stuff here, and there shouldn't be any API-aware stuff elsehwere in the frontend
 */

class FitlyApi {
    //the token for interaction with API will be stored here.
    static token;

    static async request( endpoint, data = {}, method = "get"){
        console.debug("API Call:", endpoint, data, method)

        const url = `${BASE_URL}/${endpoint}`;
        let headers;
        
        (FitlyApi.token !== "") ? headers = { Authorization: `Bearer ${FitlyApi.token}` } : headers = null;
        const params = (method === "get")
            ? data
            : {};

            try{
                return (await axios({url, method, data, params, headers})).data;
            } catch(err) {
                console.error("API Error:", err.response);
                console.log(headers)
                let message = err.response.data.error.message;
                throw Array.isArray(message) ? message : [message];
            }
            }


            //Individual Fitly API routes

            //Front End user methods

            /**
             * Get a user's information
             * 
             * Parameter: {data}
             * 
             * {data: {token}}
             * 
             * Returns: 
             *  {"user": {
             *      "username",
             *      "firstName",
             *      "lastName",
             *      "password"
             *      "email",
        *           "isAdmin"}}
             */

            static async findUser(username){
                let res = await this.request(`users/${username}`)
                return res.user
            }

        /**
         * Authenticate a user and login
         * 
         * @param {*} data
         * 
         * {data: {username, password}}
         * 
         * @returns {token}
         */

        static async login(data){   
            let res = await this.request('auth/token', data, 'post')
            return res.token;
        }

        /** 
         * Sign up a user and create an account
         * 
         * @param {*} data
         *
         * {data: {username, password}}
         * 
         * @returns {token}
         */

        static async signup(data){
            let res = await this.request(`auth/register`, data, 'post')
            return res.token;
        }

        static async updateUser(username, data){
            let res = await this.request(`users/${username}`, data, 'patch');
            return res.user;
        }

        //Workout Methods
        
        /**
         * Find all workouts in the database that a user has created
         * 
         * 
         */

        static async findAllWorkouts(){
            let res = await this.request(`workouts/`);
            return res.workouts;

        }
        
        
        /**
         * 
         * @param {*} data 
         * {data : {user_id}}
         * 
         * @returns {workout}
         */

        static async findWorkout(data){
            console.log(data)
            let res = await this.request(`workouts/${data.workoutId}`);
            return res.workout;
        }
        
        /**
         * 
         * @param {*} data 
         * {data: {user_id}}
         * @returns {workout}
         */

        static async createWorkout(data){
            let res = await this.request(`workouts/`, data, 'post');
            return res.newWorkout;
        }

        /**
         * 
         * @param {*} data 
         * {data: {user_id}}
         * @returns {workout}
         */

        static async updateWorkout(workout_id,data){
            let res = await this.request(`workouts/${workout_id}`, data, 'patch');
            return res.updatedWorkout;
        }
        
        /**
         * 
         * @param {*} data 
         * {data: {user_id}}
         * @returns {message}
         */

        static async deleteWorkout(workoutId){
            let res = await this.request(`workouts/${workoutId}`, {}, 'delete');
            return res.workout;
        }

        static async addWorkoutCircuit(data){
            let res = await this.request(`workouts//${data.workoutId}/circuits/${data.circuitId}`, data, 'post');
            return res.workoutCircuit;
        }

        //Exercise Methods

        /**
         * Find all Exercises created by a user
         * @param {*} data 
         * {data: {user_id}}
         * @returns {exercises}
         */

        static async findAllExercises(data){
            let res = await this.request(`exercises/`, data, 'get');
            return res.exercises
        }
        

        /**
         * Find a specfic exercise created by a user
         * @param {*} data 
         * {data {user_id}}
         * @returns {exercise}
         */

        static async findExercise(data){
            let res = await this.request(`exercises/${data.exercise_id}`, data, 'get');
            return res.exercise
        }
        
        /**
         * Creates a new exercise
         * @param {*} data 
         * {data: {user_id}}
         * @returns {exercise}
         */

        static async createExercise(data){
            let exercise = await this.request(`exercises/`, data, 'post');
            return exercise
        }
        
        /**
         * Update an existing exercise
         * @param {*} data 
         * {data: {exercise}}
         * @returns {exercise}
         */
        static async updateExercise(exerciseId, data){
            let res = await this.request(`exercises/${exerciseId}`, data, 'patch');
            return res.updatedExercise
        }
        
        /**
         * Delete an existing exercise 
         * @param {*} data 
         * {data: {user_id}}
         * @returns {message}
         */

        static async deleteExercise(data){
            let res = await this.request(`exercises/${data.exercise_id}`, data, 'delete');
            return res.message
        }

        //Category Methods

        /**
         * Get all default categories and categories created by the user
         * @param {*} data 
         * {data: {user_id}}
         * @returns {categories}
         */

        static async findAllCategories(data){
            let res = await this.request(`categories/`, data, 'get');
            return res.categories
        }

        /**
         * Get a category by category_id
         * @param {*} data 
         * {data: {user_id}}
         * @returns {category}
         */
        static async findCategory(data){
            let res = await this.request(`categories/${data.category_id}`, data, 'get');
            return res.category;
        }

        /**
         * Create a new category
         * @param {*} data 
         * {data: {user_id}}
         * @returns {category}
         */

        static async createCategory(data){
            let res = await this.request(`categories/`, data, 'post');
            return res.category;
        }

        /**
         * Update an existing workout category
         * @param {*} data 
         * {data: {user_id}}
         * @returns {category}
         */

        static async updateCategory(data){
            let res = await this.request(`categories/:category_id`, data, 'patch');
            return res.category;
        }

        /**
         * Delete an existing workout category
         * @param {*} data 
         * {data: {user_id}}
         * @returns {message}
         */
        static async deleteCategory(data){
            let res = await this.request(`categories/:category_id`, data, 'delete');
            return res.category;
        }

        //Equipment Methods

        /**
         * Get all default equipments and equipment created by the user
         * @param {*} data 
         * {data: {user_id}}
         * @returns {equipments}
         */

        static async findAllEquipments(data){
            let res = await this.request(`equipments/`, data, 'get');
            return res.equipments
        }

        /**
         * Get a category by category_id
         * @param {*} data 
         * {data: {user_id}}
         * @returns {equipment}
         */
        static async findEquipment(data){
            let res = await this.request(`equipments/${data.equipment_id}`, data, 'get');
            return res.equipment;
        }

        /**
         * Create a new equipment
         * @param {*} data 
         * {data: {user_id}}
         * @returns {equipment}
         */

        static async createEquipment(data){
            let res = await this.request(`equipments/`, data, 'post');
            return res.equipment;
        }

        /**
         * Update an existing workout equipment
         * @param {*} data 
         * {data: {user_id}}
         * @returns {equipment}
         */

        static async updatEquipment(data){
            let res = await this.request(`equipments/:equipment_id`, data, 'patch');
            return res.equipment;
        }

        /**
         * Delete an existing workout equipment
         * @param {*} data 
         * {data: {user_id}}
         * @returns {message}
         */
        static async deleteEquipment(data){
            let res = await this.request(`equipments/:equipment_id`, data, 'delete');
            return res.equipment;
        }

        //Circuit Methods


        static async findAllCircuits(data){
            let res = await this.request(`circuits/`, data, 'get');
            return res.circuits;
        }

        static async findCircuit(data){
            let res = await this.request(`circuits/${data.circuit_id}`, data, 'get');
            return res.circuit;
        }

        static async addCircuit(data){
            let res = await this.request(`circuits/`, data, 'post');
            return res.circuit;
        }
        
        static async updateCircuit(circuitId,data){
            let res = await this.request(`circuits/${circuitId}`, data, 'patch');
            return res.updatedCircuit;
        }

        static async deleteCircuit(circuitId){
            let res = await this.request(`circuits/${circuitId}`, {} , 'delete');
            return res.message;
        }

        static async addExerciseCircuit(data){
            let res = await this.request(`circuits/${data.circuitId}/exercises/${data.exerciseId}`, data, 'post');
            return res.circuitExercise;
        }
        
        static async updateExerciseCircuit(data){
            let res = await this.request(`circuits/${data.circuitId}/exercises/${data.exerciseId}`, data, 'patch');
            return res.circuitExercise;
        }

        static async findAllMuscleGroups(data){
            let res = await this.request(`muscleGroups/`, data, 'get');
            return res.muscleGroups
        }
        
        static async findMuscleGroup(data){
            let res = await this.request(`muscleGroups/${data.muscleGroupId}`, data, 'get');
            return res.muscleGroup
        }
    }

    FitlyApi.token = ""

export default FitlyApi;