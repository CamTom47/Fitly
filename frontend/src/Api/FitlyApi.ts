import axios, {Axios, AxiosHeaders, AxiosRequestConfig, AxiosResponse, AxiosResponseHeaders, RawAxiosRequestHeaders} from 'axios';

const BASE_URL : string = process.env.REACT_APP_BASE_URL || "http://localhost:3001";

/** Fitly API Class
 * 
 * Static class typing together methods used to get/send to the Fitly API.
 * There shouldn't be any frontend-specific stuff here, and there shouldn't be any API-aware stuff elsehwere in the frontend
 */

class FitlyApi {
    //the token for interaction with API will be stored here.
    static token: string;

    static async request( endpoint: string, data: object = {}, method: string = "get"){
        console.debug("API Call:", endpoint, data, method)

        const url: string =  `${BASE_URL}/${endpoint}`
        const params: {} = (method === "get")
        ? data
        : {}
        const headers : {} | undefined = (FitlyApi.token !== "") ? { Authorization: `Bearer ${FitlyApi.token}` } : undefined;

            try{
                const response = (await axios({ url, params, method, data, headers})).data

                return response
            } catch(err) {
                console.error("API Error:", err.response);
                let message: string = err.response.data.error.message;
                throw Array.isArray(message) ? message : [message];
            }
            }


            //Individual Fitly API routes

            //Front End user methods

            /**
             * Get a user's information
             * 
             * Parameter: {data}
             * 
             * {data: {token}}
             * 
             * Returns: 
             *  {"user": {
             *      "username",
             *      "firstName",
             *      "lastName",
             *      "password"
             *      "email",
        *           "isAdmin"}}
             */

            static async findUser<Promise>(username: string) : Promise {
                console.log(username)
                let res = await this.request(`users/${username}`)
                return res.user
            }

        /**
         * Authenticate a user and login
         * 
         * @param {*} data
         * 
         * {data: {username, password}}
         * 
         * @returns {token}
         */

        static async login<Promise>(data: {}) : Promise{   
            console.log(data)
            let res = await this.request('auth/token', data, 'post')
            return res.token;
        }

        /** 
         * Sign up a user and create an account
         * 
         * @param {*} data
         *
         * {data: {username, password}}
         * 
         * @returns {token}
         */

        static async signup<Promise>(data: {}) : Promise{
            let res = await this.request(`auth/register`, data, 'post')
            return res.token;
        }

        static async updateUser<Promise>(username: string, data : {}) : Promise{
            let res = await this.request(`users/${username}`, data, 'patch');
            return res.user;
        }

        //Workout Methods
        
        /**
         * Find all workouts in the database that a user has created
         * 
         * 
         */

        static async findAllWorkouts<Promise>(data? : {}) : Promise{
            let res = await this.request(`workouts/`, data, 'get');
            return res.workouts;

        }
        
        
        /**
         * 
         * @param {*} data 
         * {data : {user_id}}
         * 
         * @returns {workout}
         */

        static async findWorkout<Promise>(data: {workoutId: number}) : Promise{
            console.log(data)
            let res = await this.request(`workouts/${data.workoutId}`);
            return res.workout;
        }
        
        /**
         * 
         * @param {*} data 
         * {data: {user_id}}
         * @returns {workout}
         */

        static async createWorkout<Promise>(data : {}) : Promise{
            console.log(data)
            let res = await this.request(`workouts/`, data, 'post');
            return res.newWorkout;
        }

        /**
         * 
         * @param {*} data 
         * {data: {user_id}}
         * @returns {workout}
         */

        static async updateWorkout<Promise>(workoutId: number ,data : {}) : Promise{
            let res = await this.request(`workouts/${workoutId}`, data, 'patch');
            return res.updatedWorkout;
        }
        
        /**
         * 
         * @param {*} data 
         * {data: {user_id}}
         * @returns {message}
         */

        static async deleteWorkout<Promise>(workoutId: number) : Promise{
            let res = await this.request(`workouts/${workoutId}`, {}, 'delete');
            return res.workout;
        }

        static async addWorkoutCircuit<Promise>(data : {workoutId: number | undefined, circuitId: number | undefined}) : Promise{
            let res = await this.request(`workouts//${data.workoutId}/circuits/${data.circuitId}`, data, 'post');
            return res.workoutCircuit;
        }

        //Exercise Methods

        /**
         * Find all Exercises created by a user
         * @param {*} data 
         * {data: {user_id}}
         * @returns {exercises}
         */

        static async findAllExercises<Promise>(data?: {}) : Promise{
            let res = await this.request(`exercises/`, data, 'get');
            return res.exercises
        }
        

        /**
         * Find a specfic exercise created by a user
         * @param {*} data 
         * {data {user_id}}
         * @returns {exercise}
         */

        static async findExercise<Promise>(data : {exerciseId : number}): Promise{
            let res = await this.request(`exercises/${data.exerciseId}`, data, 'get');
            return res.exercise
        }
        
        /**
         * Creates a new exercise
         * @param {*} data 
         * {data: {user_id}}
         * @returns {exercise}
         */

        static async createExercise<Promise>(data: {}): Promise{
            let exercise = await this.request(`exercises/`, data, 'post');
            return exercise
        }
        
        /**
         * Update an existing exercise
         * @param {*} data 
         * {data: {exercise}}
         * @returns {exercise}
         */
        static async updateExercise<Promise>(exerciseId : number, data: {}): Promise{
            let res = await this.request(`exercises/${exerciseId}`, data, 'patch');
            return res.updatedExercise
        }
        
        /**
         * Delete an existing exercise 
         * @param {*} data 
         * {data: {user_id}}
         * @returns {message}
         */

        static async deleteExercise<Promise>(data : {exerciseId: number}): Promise{
            let res = await this.request(`exercises/${data.exerciseId}`, data, 'delete');
            return res.message
        }

        //Category Methods

        /**
         * Get all default categories and categories created by the user
         * @param {*} data 
         * {data: {user_id}}
         * @returns {categories}
         */

        static async findAllCategories<Promise>(data? : {}) : Promise{
            let res = await this.request(`categories/`, data, 'get');
            return res.categories
        }

        /**
         * Get a category by category_id
         * @param {*} data 
         * {data: {user_id}}
         * @returns {category}
         */
        static async findCategory<Promise>(data : {categoryId : string}) : Promise{
            let res = await this.request(`categories/${data.categoryId}`, data, 'get');
            return res.category;
        }

        /**
         * Create a new category
         * @param {*} data 
         * {data: {user_id}}
         * @returns {category}
         */

        static async createCategory<Promise>(data : {}) : Promise{
            let res = await this.request(`categories/`, data, 'post');
            return res.category;
        }

        /**
         * Update an existing workout category
         * @param {*} data 
         * {data: {user_id}}
         * @returns {category}
         */

        static async updateCategory<Promise>(data : {categoryId : number,
                                                    name: string,
                                                    description: string
        }) : Promise{
            let res = await this.request(`categories/${data.categoryId}`, data, 'patch');
            return res.category;
        }

        /**
         * Delete an existing workout category
         * @param {*} data 
         * {data: {user_id}}
         * @returns {message}
         */
        static async deleteCategory<Promise>(data : {categoryId : number}) : Promise{
            let res = await this.request(`categories/${data.categoryId}`, data, 'delete');
            return res.category;
        }

        //Equipment Methods

        /**
         * Get all default equipments and equipment created by the user
         * @param {*} data 
         * {data: {user_id}}
         * @returns {equipments}
         */

        static async findAllEquipments<Promise>(data? : {}) : Promise{
            let res = await this.request(`equipments/`, data, 'get');
            return res.equipments
        }

        /**
         * Get a category by category_id
         * @param {*} data 
         * {data: {user_id}}
         * @returns {equipment}
         */
        static async findEquipment<Promise>(data : {equipmentId : number}) : Promise{
            let res = await this.request(`equipments/${data.equipmentId}`, data, 'get');
            return res.equipment;
        }

        /**
         * Create a new equipment
         * @param {*} data 
         * {data: {user_id}}
         * @returns {equipment}
         */

        static async createEquipment<Promise>(data : {}) : Promise{
            let res = await this.request(`equipments/`, data, 'post');
            return res.equipment;
        }

        /**
         * Update an existing workout equipment
         * @param {*} data 
         * {data: {user_id}}
         * @returns {equipment}
         */

        static async updatEquipment<Promise>(data : {equipmentId}) : Promise{
            let res = await this.request(`equipments/${data.equipmentId}`, data, 'patch');
            return res.equipment;
        }

        /**
         * Delete an existing workout equipment
         * @param {*} data 
         * {data: {user_id}}
         * @returns {message}
         */
        static async deleteEquipment<Promise>(data : {equipmentId : number}) : Promise{
            let res = await this.request(`equipments/${data.equipmentId}`, data, 'delete');
            return res.equipment;
        }

        //Circuit Methods


        static async findAllCircuits<Promise>(data? : {}) : Promise{
            let res = await this.request(`circuits/`, data, 'get');
            return res.circuits;
        }

        static async findCircuit<Promise>(data : {circuitId}) : Promise{
            let res = await this.request(`circuits/${data.circuitId}`, data, 'get');
            return res.circuit;
        }

        static async addCircuit<Promise>(data : {}) : Promise{
            let res = await this.request(`circuits/`, data, 'post');
            return res.circuit;
        }
        
        static async updateCircuit<Promise>(circuitId : number, data : {}) : Promise{
            let res = await this.request(`circuits/${circuitId}`, data, 'patch');
            return res.updatedCircuit;
        }

        static async deleteCircuit<Promise>(circuitId : number) : Promise{
            let res = await this.request(`circuits/${circuitId}`, {} , 'delete');
            return res.message;
        }

        static async addExerciseCircuit<Promise>(data : {circuitId : number | undefined, exerciseId : number | undefined}) : Promise{
            let res = await this.request(`circuits/${data.circuitId}/exercises/${data.exerciseId}`, data, 'post');
            return res.circuitExercise;
        }
        
        static async updateExerciseCircuit<Promise>(data : {circuitId: number | undefined, exerciseId: number | undefined}) : Promise{
            let res = await this.request(`circuits/${data.circuitId}/exercises/${data.exerciseId}`, data, 'patch');
            return res.circuitExercise;
        }

        static async findAllMuscleGroups<Promise>(data? : {}) : Promise{
            let res = await this.request(`muscleGroups/`, data, 'get');
            return res.muscleGroups
        }
        
        static async findMuscleGroup<Promise>(data : {muscleGroupId : number}) : Promise{
            let res = await this.request(`muscleGroups/${data.muscleGroupId}`, data, 'get');
            return res.muscleGroup
        }
    }

    FitlyApi.token = ""

export default FitlyApi;