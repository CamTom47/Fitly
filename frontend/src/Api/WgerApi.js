import axios from 'axios';

const WGER_BASE_URL = "https://wger.de/api/v2"

    class WgerApi {
        static async request( endpoint, data = {}, method = "get"){
            console.debug("WGER Call:", endpoint, data, method)

            const url = `${WGER_BASE_URL}/${endpoint}`

            const params = (method === "get")
            ? data
            : {};
            try{
                const res = (await axios({url, method, data, params}));
                return res
            } catch(err){
                console.error("wger API Error:", err.response)
                let message = err.response.data.error.message
                throw Array.isArray(message) ? message : [message];
            }
        }

        static async getAllExercises(endpoint = 'exercisebaseinfo/'){
            if(endpoint.includes(WGER_BASE_URL)){
                endpoint = endpoint.split(`${WGER_BASE_URL}/`)[1]
            };
            
            console.log(endpoint)
            let res = await this.request(`${endpoint}`)
            console.log(res)
            return res
        }

    }


    export default WgerApi