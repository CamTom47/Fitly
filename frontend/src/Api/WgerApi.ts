import axios, { AxiosResponse } from 'axios';

const WGER_BASE_URL : string = "https://wger.de/api/v2"

    class WgerApi {
        static async request( endpoint: string, data: object = {}, method: string = "get"){
            console.debug("WGER Call:", endpoint, data, method)

            const url: string = `${WGER_BASE_URL}/${endpoint}`

            const params: {} = (method === "get")
            ? data
            : {};
            try{
                const res = (await axios({url, method, data, params}));
                return res
            } catch(err){
                console.error("wger API Error:", err.response)
                let message : string = err.response.data.error.message
                throw Array.isArray(message) ? message : [message];
            }
        }

        static async getAllExercises<Promise>(endpoint : string = 'exercisebaseinfo/'): Promise {

            // The initial request will set the next and previous calls with a end point similar to 
            // "next": "https://wger.de/api/v2/exercisebaseinfo/?limit=20&offset=20",
            // Splitting on the WGER_BASE_URL allows for the new and point to be added to the default WGER_BASE_URL

            if(endpoint.includes(WGER_BASE_URL)){
                endpoint = endpoint.split(`${WGER_BASE_URL}/`)[1]
            };
            
            let res = await this.request(endpoint)

            return res.data
        }

    }


    export default WgerApi