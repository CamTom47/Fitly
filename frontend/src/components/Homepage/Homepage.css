.HomepageContainer {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 75%;
  margin: 0 auto;
}

.HomepageContainer h1 {
  margin: 7rem 0 0;
  font-size: 7rem;
  font-weight: 500;
  color: var(--white);
  text-align: center;
}
.HomepageText{
  color: var(--white);
  text-align: center;
  margin-top: 2rem;
  font-size: 20px;
}

#cardContainer{
  display: grid;
  grid-template-columns: [start] 20rem [line1] 20rem [line2] 20rem [end]; 
  column-gap: 5rem;
  margin-top: 7rem;
  padding-bottom: 5rem;
}
