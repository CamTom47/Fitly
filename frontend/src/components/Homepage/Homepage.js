import React, { useContext} from "react";
import LoginForm from "../Forms/LoginForm/LoginForm";
import UserContext from "../../context/UserContext";
import { Card } from "reactstrap";

const Homepage = () => {

    const { currentUser, login } = useContext(UserContext);

    if(!currentUser) 
    return(
        <div className="d-flex justify-content-center m-5 ">
            <Card className="d-flex flex-column align-items-center p-5">
                <LoginForm login={login}></LoginForm>
                <hr className="w-75"></hr>
                <p>New to Fitly?</p><a href="/register">Sign Up Here</a>
            </Card>
        </div>
    )
}

export default Homepage;