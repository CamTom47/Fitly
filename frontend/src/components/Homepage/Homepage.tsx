import React from "react";
import LoginForm from "../Forms/LoginForm/LoginForm";
import { Card } from "reactstrap";
import { selectCurrentUser } from '../../slices/usersSlice';
import { useAppSelector} from '../../hooks/reduxHooks'

const Homepage = (): React.JSX.Element  => {
    
    const currentUser = useAppSelector(selectCurrentUser);

    return !currentUser ? 
        (
                <LoginForm></LoginForm>
        ) 
    :
        (
            <h1>User Dashboard In Progress</h1>
        )
    
}

export default Homepage;