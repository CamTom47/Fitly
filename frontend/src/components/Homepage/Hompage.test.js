import { render, waitFor} from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
// import userEvent from '@testing-library/user-event'
import FitlyApi from '../../Api/FitlyApi';
import WgerApi from '../../Api/WgerApi';
import UserProvider from '../../../testUtils';
import Homepage from './Homepage';
 
//Mock FitlyAPI
jest.mock('../../Api/FitlyApi', () => ({
    findAllExercises: jest.fn(() => {

    })
}));

//Mock WgerApi
jest.mock('../../Api/WgerApi', () => {
});



describe('Homepage Functionality', () => {

    let children = (
        <Homepage/>
    );
    
  it('it renders without crashing', async () => {

    render(
        <UserProvider children={children}/>
      );
    }); 
    
    it('matches snapshot', async () => {
        const {asFragment} = render(
    <UserProvider children={children}/>
)
     expect(asFragment()).toMatchSnapshot();
    })
});