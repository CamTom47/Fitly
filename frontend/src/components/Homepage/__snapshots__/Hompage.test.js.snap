// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage Functionality matches snapshot 1`] = `<DocumentFragment />`;
